seccon23-rev-perfect-blu/index.md

```diff
@@ -0,0 +1,403 @@
+
+There a multiple types of DVD menus that I'm aware of:
+- Java menus
+- IGS Menus
+
+What I did first is I checked for any .jar files or just something that could remind me of Java, but I didn't find anything and it meant that I was dealing with the IGS menu.
+
+The first step in IGS menu analysis is to download [BDEdit](https://bdedit.pel.hu/) and open up our mounted `BDMV\index.bdmv`
+
+![2](./img/2.png)
+
+Looks scary, I know. What I did is, I checked the clip info for the `und` streams. To do this I clicked on the `CLIPINFO` menu and found the `und` stream in the GUI that contained our buttons logic.
+
+![3](./img/3.png)
+
+Here, at the left top corner, there's a combo box with a clip selector, there are 96 clips and all of them have some buttons. 
+
+When I double-clicked at the `und` stream, the menu with buttons opened up and I saw a lot of buttons and the disasm of the code that they were doing.
+
+![4](./img/4.png)
+
+By messing around and guessing I found that the first default valid button(`1FDE` on the screenshot) contains some random stuff that I wasn't interested in, and I should check others instead.
+
+![13](./img/13.png)
+
+In this bytecode there's a `Call Object **` instruction with opcode `21820000` and only one operand, that just starts playing another menu that you provide as its first operand. Knowing this, I started analysing what the other buttons were doing.
+
+
+By observing all the other buttons I saw three types of buttons.
+
+---
+
+Type 1, what most buttons are doing.
+
+![6](./img/6.png)
+
+This button jumps from our current menu(`0`) to the menu `48`.
+
+---
+
+Type 2.
+
+![7](./img/7.png)
+
+This button jumps from our current menu(`0`) to the next menu (`1`).
+
+---
+
+Type 3.
+
+![8](./img/8.png)
+
+This button is jumping to clip 96, which I checked by playing the `BDMV\STREAM\00096.m2ts` file in VLC.
+
+![9](./img/9.png)
+
+---
+
+After that, I knew that there were only three destinations on the first menu
+* Clip 1
+* Clip 48
+* Clip 96 (`WRONG...`) 
+
+When I checked the menu buttons for clip 1 the pattern looked still the same so I checked menu 48 instead, and on clip 48 all the menu buttons were doing the same thing, jumping to menu 48 (or 96).
+
+Seems odd, huh? I investigated it a bit and it seemed like we would always end up on the menu 96 if we are on the menu that's index is >= 48.
+
+There's one exception though, that I just guessed. Remember how I opened stream 96 in vlc? Well, I did the same thing for clip 95 and got this.
+
+![10](./img/10.png)
+
+From now on solving this challenge looked trivial, I needed to parse all the clips and find what buttons were leading us to clip 95.
+
+#### Solving
+
+While the idea was easy enough, I struggled for half an hour trying to parse the clips.
+
+I tried many things/libs to parse the clips and extract this bytecode from them but none of them worked so I decided that i should just do it by myself.
```

Comment:
_I tried many things_ - what things and why not just "libraries"
also small "i" and extra "by"

```
I tried a bunch of libraries to parse the clips and extract the bytecode from them.
However, none of them worked, so I decided I should just do it myself.
```